sdk/tests/client/input_selection/alias_outputs.rs

```diff
@@ -2291,3 +2306,100 @@ fn sender_in_governor_but_not_owned() {
         Err(Error::UnfulfillableRequirement(Requirement::Sender(sender))) if sender == Address::try_from_bech32(BECH32_ADDRESS_ED25519_1).unwrap()
     ));
 }
+
+#[test]
+fn new_state_metadata() {
+    let protocol_parameters = protocol_parameters();
+    let alias_id_1 = AliasId::from_str(ALIAS_ID_1).unwrap();
+
+    let alias_output =
+        AliasOutputBuilder::new_with_minimum_storage_deposit(*protocol_parameters.rent_structure(), alias_id_1)
+            .with_state_metadata(vec![1, 2, 3])
+            .add_unlock_condition(StateControllerAddressUnlockCondition::new(
+                Address::try_from_bech32(BECH32_ADDRESS_ED25519_0).unwrap(),
+            ))
+            .add_unlock_condition(GovernorAddressUnlockCondition::new(
+                Address::try_from_bech32(BECH32_ADDRESS_ED25519_0).unwrap(),
+            ))
+            .finish_output(protocol_parameters.token_supply())
+            .unwrap();
+
+    let inputs = vec![InputSigningData {
+        output: alias_output.clone(),
+        output_metadata: OutputMetadata::new(rand_block_id(), rand_output_id(), false, None, None, None, 0, 0, 0),
+        chain: None,
+    }];
+
+    // New alias output, with updated state index
+    let updated_alias_output = AliasOutputBuilder::from(alias_output.as_alias())
+        .with_minimum_storage_deposit(*protocol_parameters.rent_structure())
+        .with_state_metadata(vec![3, 4, 5])
+        .with_state_index(alias_output.as_alias().state_index() + 1)
+        .finish_output(protocol_parameters.token_supply())
+        .unwrap();
+
+    let outputs = vec![updated_alias_output];
+
+    let selected = InputSelection::new(
+        inputs.clone(),
+        outputs.clone(),
+        addresses(vec![BECH32_ADDRESS_ED25519_0]),
+        protocol_parameters,
+    )
+    .required_inputs(HashSet::from_iter([*inputs[0].output_id()]))
```

Comment:
Just checking, is this necessary? You have the alias as output so requirement will make the input mandatory anyway?